packages/usa-input-mask/src/index.js

```diff
@@ -0,0 +1,128 @@
+const selectOrMatches = require("../../uswds-core/src/js/utils/select-or-matches");
+const behavior = require("../../uswds-core/src/js/utils/behavior");
+const { prefix: PREFIX } = require("../../uswds-core/src/js/config");
+const { CLICK } = require("../../uswds-core/src/js/events");
+
+const MASKED_CLASS = `${PREFIX}-masked`;
+const MASKED = `.${MASKED_CLASS}`;
+const MASK = `${PREFIX}-input-mask`;
+const MASK_CONTENT = `${MASK}--content`;
+const PLACEHOLDER = "placeholder";
+
+// User defined Values
+const maskedNumber = "_dDmMyY9";
+const maskedLetter = "_";
+
+// replaces each masked input with a shell containing the input and it's mask.
+const createMaskedInputShell = (input) => {
+  const placeholder = input.getAttribute(`${PLACEHOLDER}`);
+  if (placeholder) {
+    input.setAttribute("maxlength", placeholder.length);
+    input.setAttribute("data-placeholder", placeholder);
+    input.removeAttribute(`${PLACEHOLDER}`);
+  } else {
+    return;
+  }
+
+  const shell = document.createElement("span");
+  shell.classList.add(MASK);
+  shell.setAttribute("data-mask", placeholder);
+  console.log("shell ", shell); // eslint-disable-line no-console
+
+  const content = document.createElement("span");
+  content.classList.add(MASK_CONTENT);
+  content.setAttribute("aria-hidden", "true");
+  content.id = `${input.id}Mask`;
+  content.textContent = placeholder;
+
+  shell.appendChild(content);
+  input.parentNode.insertBefore(shell, input);
+  shell.appendChild(input);
+};
+
+const setValueOfMask = (el) => {
+  const { value } = el;
+  const placeholder = el.getAttribute("data-placeholder");
+  const placeholderVal = `${placeholder.substr(value.length)}`;
+
+  const theIEl = document.createElement("i");
+  theIEl.textContent = value;
+  return [theIEl, placeholderVal];
+};
+
+const handleCurrentValue = (el) => {
+  const isCharsetPresent = el.getAttribute("data-charset");
```

Comment:
data-charset is for alphanumeric values, like in a Canadian ZIP code (which we are currently not using, but it's there as an option). We could probably name this better? like data-alphanumeric what do you think? Also yes this needs to be added to the guidance! Thanks

Reply:
Sorry if this is a silly question are the options strictly alpha, numeric, or both (alphanumeric)? The values would be a regex that developer sets?

I think the name could be flexible, but the most important thing is that it's documented. Perhaps `data-charset-type`.

Reply:
@mejiaj When the data-charset attribute is added, it's an alphanumeric placeholder that should match the regex in the pattern attribute. NOTE: If data-charset is not present, the input is strictly numeric.

The alphanumeric placeholder would take the form of X for letter and _ for number (this is customizable in the JS)

So for a Canadian postal code (*see description below), it would be:
"pattern": "\w\d\w \d\w\d",
"data-charset": "X_X _X_"

Let me know if that makes sense... I'm discussing with the team now if we should add a fourth example that shows some alphanumeric input masking, and if so what example would that be... Do you have any thoughts on that? Thanks!

* A Canadian postal code is a six-character uniformly structured, alphanumeric code in the form “ANA NAN” where “A” is an alphabetic character and “N” is a numeric character.

Reply:
Small correction on the above... Just talked with Dan and the data-charset character placeholders should be # for number and A for letter like this `A#A #A#`. I'm going to be updating this in the code.